DiagnosticsExtension/Controllers/ConnectionStringValidationController.cs

```diff
@@ -7,41 +7,46 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
 using System.Net.Sockets;
 using System.Text;
 using System.Threading.Tasks;
+using System.Web;
 using System.Web.Http;
+using DiagnosticsExtension.Models;
 using DiagnosticsExtension.Models.ConnectionStringValidator;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace DiagnosticsExtension.Controllers
 {
-    /// <summary>
-    /// Worker instances are running an udp echo server on port 30000. This controller is for checking the connection between target 
-    /// worker instance by pinging and checking the echoed result.
-    /// </summary>
     [RoutePrefix("api/connectionstringvalidation")]
     public class ConnectionStringValidationController : ApiController
     {
         [HttpGet]
         [Route("validate")]
         public async Task<HttpResponseMessage> Validate(string connStr, int? typeId = null)
         {
-            // register all validators here
-            var typeValidatorMap = new IConnectionStringValidator[]
+            // register all validators here, the order of validators decides the order of connection string matching
+            var validators = new IConnectionStringValidator[]
```

Comment:
Move this to class member and initialize in constructor

Reply:
done